crates/runtime/src/component/catalog.rs

```diff
@@ -21,7 +21,7 @@ use spicepod::{component::catalog as spicepod_catalog, param::Params};
 use std::{collections::HashMap, sync::Arc};
 
 use super::{find_first_delimiter, validate_identifier};
-use crate::Runtime;
+use crate::{Runtime, component::dataset::AccessMode};
```

Comment:
Given this is now common across datasets and catalogs, let's pull it out into a common module.

Reply:
Good point, thank you, confirm I'll do this as part of next PR with additional integration tests for catalog support